module/zfs/spa.c

```diff
@@ -7554,6 +7554,7 @@ spa_vdev_split_mirror(spa_t *spa, char *newname, nvlist_t *config,
 	spa_history_log_internal(newspa, "split", NULL,
 	    "from pool %s", spa_name(spa));
 
+	newspa->spa_config_source = SPA_CONFIG_SRC_NONE;
```

Comment:
This makes sense, but I'm concerned that we're leaving a newly split pool in an impossible state.  The `spa_load_impl()` function clearly asserts at the top that every imported pool needs to have a `spa_config_source != SPA_CONFIG_SRC_NONE`.  Yet after a pool split this is exactly the state we're leaving the new pool in.

At the moment, this is harmless but I think it would be preferable to add a new boolean specifically for this.  How about `boolean_t spa->spa_is_splitting` which can be set for the duration of the split.  It can be declared after `spa_is_exporting` in the structure.  Can you also add a comment either explaining why this is needed, or referencing the relevant FreeBSD code and comment where this exception is handled.

Reply:
That's actually what is done in FreeBSD now. I just thought this was cleaner since nothing actually checks this value later on.

Reply:
That's true, nothing checks it today but that might not always be the case.  I'd prefer we add the new `spa->spa_is_splitting` boolean which is set when `SPA_CONFIG_SRC_SPLIT` is set above and cleared here.  Then you can use it the same way in the platform code.